### ISSUE TITLE
No indication in web UI that hashtag search pinned columns only work for the first four hashtags

### ISSUE THREAD

**Issue by Cassolotl** [tootsuite/mastodon#12458]
Someone opened *Regression: Pinned hashtag feature does not work anymore with multiple hashtags #12453*, because tags they had added to their pinned hashtag search column weren't appearing.

It turns out that this is working as intended, and nothing beyond the first four hashtags was meant to show up at all.

Gargron [said](https://github.com/tootsuite/mastodon/issues/12453#issuecomment-557809857):

>You can combine at most 4 hashtags per mode (any/all/none) since v3.0.1 for performance reasons.

### Expected behaviour

It should not be possible to enter more hashtags.

### Actual behaviour

When you enter more than four hashtags in a pinned search column, toots containing hashtags beyond the first four are entirely omitted, without warning or error message.

### Steps to reproduce the problem

1. Search hashtag.
2. Pin hashtag search column.
3. Click column's "settings" icon, and check "Include additional tags for this column".
4. Enter more than four hashtags, and note the lack of error message, note that it is possible to enter more than four hashtags at all.
5. Note that there are no toots in the column containing hashtags beyond the first four, with no warning or error message.

### Specifications

Mastodon 3.0.1

**Comment by marrus-sh** [comment]
It would be nice if there was some kind of notification/error for existing users with more than 4 hashtags per column which did NOT automatically delete extra hashtags, so that they could make decisions about which hashtags to keep themselves.

**Comment by rohieb** [comment]
Yes, this would be nice, as it would clear up confusion. Also in other apps like Tusky, but that's a different story … :)

**Comment by Cassolotl** [comment]
Is this still an issue?

**Comment by rohieb** [comment]
In the web interface, I can now add four other hashtags to the "any of these" input box, which is (I think) one more than before, and see posts for all five hashtags in the column. When I try to enter an additional hashtag, it doesn't appear in the input box after I choose it from the autocompletion, which is consistent, but I still find it a bit confusing that there is no explanation about the maximum number of hashtags.

**Comment by Nicd** [comment]
Just encountered this (I think) on my own instance. The column itself obviously searches for one hashtag, and I can add up to 4 additional hashtags in "any of these", but the last hashtag in that list does not affect anything. So I suppose the count of four is already filled with the column hashtag + three of the first in the list and the list mistakenly lets you add a fourth.

**Comment by Cassolotl** [comment]
Just testing this on mastodon.social. When you try to add a fifth hashtag to "Any of these" in the pinned hashtag column, the hashtag doesn't appear when you click or press enter, and there is no error message.

Also, posts from the last hashtag entered are not included in the pinned hashtag column, even though you are permitted to enter it. Is this because the count of four hashtags doesn't include the first hashtag you entered in order to pin the column in the first place?
